nixos/modules/services/security/proxychains.nix

```diff
@@ -0,0 +1,184 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+
+  cfg = config.services.proxychains;
+
+  configFile = ''
+    ${cfg.chain.type}
+    ${optionalString (cfg.chain.type == "random")
+    "chain_len = ${builtins.toString cfg.chain.length}"}
+    ${optionalString cfg.proxyDNS "proxy_dns"}
+    ${optionalString cfg.quietMode "quiet_mode"}
+    ${cfg.remoteDNSSubnet}
+    ${cfg.tcpReadTimeOut}
+    ${cfg.tcpConnectTimeOut}
+    ${cfg.localnet}
+    ${cfg.proxies}
+    ${cfg.extraConfig}
+  '';
+
+  proxyOptions = {
```

Comment:
Looking at the [sample config file](https://github.com/haad/proxychains/blob/master/src/proxychains.conf) (which I think would be great to put in proxychains `$out`), you can also pass a password, which is missing here yet.

Reply:
Done via f9bdbf2a13d9db1f51f7b361cb3de7cb68bea9ab

Reply:
I've deliberately decided to not include `user` and `password` proxy options since it seems pretty obscure (the most common use-case is just running this with local Tor proxy) and poses a problem with passwords-in-store.